contracts/test/sdk/IpcContract.t.sol

```diff
@@ -149,36 +172,64 @@ contract IpcExchangeTest is Test, IntegrationTestBase {
         vm.startPrank(gateway);
         //
         // an unrecognized result
-        envelope.kind = IpcMsgKind.Result;
-        envelope.message = abi.encode(ResultMsg({outcome: OutcomeType.Ok, id: keccak256("foo"), ret: bytes("")}));
+        callEnvelope.kind = IpcMsgKind.Result;
+        callEnvelope.message = abi.encode(ResultMsg({outcome: OutcomeType.Ok, id: keccak256("foo"), ret: bytes("")}));
 
-        IPCAddress memory from = envelope.from;
-        envelope.from = envelope.to;
-        envelope.to = from;
+        IPCAddress memory from = callEnvelope.from;
+        callEnvelope.from = callEnvelope.to;
+        callEnvelope.to = from;
 
         vm.expectRevert(IpcHandler.UnrecognizedResult.selector);
-        exch.handleIpcMessage(envelope);
+        exch.handleIpcMessage(callEnvelope);
     }
 
     function test_IpcExchangeTestReceiptCorrelation() public {
         vm.startPrank(gateway);
-        vm.mockCall(gateway, abi.encodeWithSelector(IGateway.sendContractXnetMessage.selector), abi.encode(envelope));
+        vm.mockCall(
+            gateway,
+            abi.encodeWithSelector(IGateway.sendContractXnetMessage.selector),
+            abi.encode(callEnvelope)
+        );
         vm.deal(address(this), 1000);
-        exch.performIpcCall_(envelope.from, CallMsg({method: bytes("1234"), params: bytes("AABB")}), 1);
+        exch.performIpcCall_(callEnvelope.from, CallMsg({method: bytes("1234"), params: bytes("AABB")}), 1);
 
-        // we store the correct envelope in the correlation map.
-        IpcEnvelope memory correlated = exch.getInflight(envelope.toHash());
-        require(correlated.toHash() == envelope.toHash());
+        // we store the correct callEnvelope in the correlation map.
+        IpcEnvelope memory correlated = exch.getInflight(callEnvelope.toHash());
+        require(correlated.toHash() == callEnvelope.toHash());
 
         // TODO test receipt correlation
 
         // TODO test dropMessages
     }
+
+    function test_IpcExchangeCallResult() public {
+        vm.mockCall(
+            gateway,
+            abi.encodeWithSelector(IGateway.sendContractXnetMessage.selector),
+            abi.encode(callEnvelope)
+        );
+        vm.deal(address(this), 1000);
+        exch.performIpcCall_(ipcAddressA, callMsg, 1);
+
+        //possibly move start prank here
+        vm.startPrank(gateway);
+        exch.handleIpcMessage(callEnvelope);
+
+        // succeeds.
+        IpcEnvelope memory lastEnvelope = exch.getLastEnvelope();
+        CallMsg memory lastCall = exch.getLastCallMsg();
+        require(keccak256(abi.encode(callEnvelope)) == keccak256(abi.encode(lastEnvelope)), "unexpected callEnvelope");
+        require(keccak256(abi.encode(callMsg)) == keccak256(abi.encode(lastCall)), "unexpected callmsg");
+
+        // gateway calls callback with result
+        exch.handleIpcMessage(resultEnvelope);
+        require(exch.handleIpcResultCalled(), "_handleIpcResult was not called");
```

Comment:
The way I understand this: we'd like `_handleIpcResult` to actually _be called_ but we are adding this test to demonstrate that it _isn't called_ . Otherwise we'd have a failing PR check and wouldn't be able to merge the PR. This also what the test name `test_IpcExchangeCallResult` suggested. 
Correct?